does update work or is the Controller in kubernetes-api now broken?

see this when folks use mvn fabric8:apply....
```
mvn fabric8:apply
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option PermSize=256m; support was removed in 8.0
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=512m; support was removed in 8.0
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building Jenkins 2.3-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- fabric8-maven-plugin:2.2.23:apply (default-cli) @ jenkins ---
[INFO] Using kubernetes at: https://vagrant.f8:8443/api/v1/ in namespace default
[INFO] Kubernetes JSON: /Users/chmoulli/Fuse/Fuse-projects/fabric8/fabric8-devops/jenkins/target/classes/kubernetes.json
[INFO] Is OpenShift: true
[INFO] Creating a namespace default
[INFO] No property defined for template parameter: fabric8.apply.JENKINS_GOGS_USER
[INFO] No property defined for template parameter: fabric8.apply.JENKINS_WORKFLOW_GIT_REPOSITORY
[INFO] No property defined for template parameter: fabric8.apply.DOMAIN
[INFO] No property defined for template parameter: fabric8.apply.GPG_PASSPHRASE
[INFO] No property defined for template parameter: fabric8.apply.JENKINS_GOGS_EMAIL
[INFO] No property defined for template parameter: fabric8.apply.JENKINS_GOGS_PASSWORD
[INFO] No property defined for template parameter: fabric8.apply.SONATYPE_USERNAME
[INFO] No property defined for template parameter: fabric8.apply.DOCKER_REGISTRY_PASSWORD
[INFO] No property defined for template parameter: fabric8.apply.DOCKER_REGISTRY_USERNAME
[INFO] No property defined for template parameter: fabric8.apply.JENKINS_SLAVE_IMAGE
[INFO] No property defined for template parameter: fabric8.apply.SONATYPE_PASSWORD
[INFO] No property defined for template parameter: fabric8.apply.FABRIC8_DEFAULT_ENVIRONMENTS
[INFO] No property defined for template parameter: fabric8.apply.SEED_GIT_URL
[INFO] Creating a template from kubernetes.json namespace default name jenkins
[INFO] Created template: jenkins/target/fabric8/applyJson/default/template-jenkins.json
[INFO] Updating a service account from kubernetes.json
[INFO] Updated service account: jenkins/target/fabric8/applyJson/default/serviceaccount-jenkins.json
[INFO] Updating a service from kubernetes.json
[ERROR] Failed to update controller from kubernetes.json. io.fabric8.kubernetes.client.KubernetesClientException: Cannot update read-only resources. Service(apiVersion=v1, kind=Service, metadata=ObjectMeta(annotations={}, creationTimestamp=null, deletionTimestamp=null, generateName=null, generation=null, labels={provider=fabric8, component=jenkins}, name=jenkins, namespace=null, resourceVersion=null, selfLink=null, uid=null, additionalProperties={}), spec=ServiceSpec(clusterIP=null, deprecatedPublicIPs=[], portalIP=null, ports=[ServicePort(name=null, nodePort=null, port=80, protocol=TCP, targetPort=IntOrString(IntVal=8080, Kind=null, StrVal=null, additionalProperties={}), additionalProperties={})], selector={provider=fabric8, component=jenkins}, sessionAffinity=null, type=LoadBalancer, additionalProperties={}), status=null, additionalProperties={})
io.fabric8.kubernetes.client.KubernetesClientException: Cannot update read-only resources
```
Either that or maybe the Controller code in kubernetes-api needs some work?

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java
@@ -130,9 +130,4 @@ public class ReplicationControllerOperationsImpl extends HasMetadataOperation<Ku
     }
     return new RollingUpdater(getClient()).rollUpdate(get(), rc);
   }
-
-  @Override
-  public ReplicationController update(ReplicationController item) {
-    return replace(item);
-  }
 }

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ServiceOperationsImpl.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ServiceOperationsImpl.java
@@ -23,7 +23,7 @@ import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.KubernetesClientException;
 import io.fabric8.kubernetes.client.dsl.ClientResource;
 
-public class ServiceOperationsImpl extends BaseOperation<KubernetesClient, Service, ServiceList, DoneableService, ClientResource<Service, DoneableService>> {
+public class ServiceOperationsImpl extends HasMetadataOperation<KubernetesClient, Service, ServiceList, DoneableService, ClientResource<Service, DoneableService>> {
 
   public ServiceOperationsImpl(KubernetesClient client) {
     this(client, null, null, true, null);
@@ -37,11 +37,7 @@ public class ServiceOperationsImpl extends BaseOperation<KubernetesClient, Servi
   public Service replace(Service item) {
       try {
         Service old = get();
-        String resourceVersion = old.getMetadata().getResourceVersion();
-        return handleReplace(getResourceUrl(), new ServiceBuilder(item)
-          .editMetadata()
-          .withResourceVersion(resourceVersion)
-          .endMetadata()
+        return super.replace(new ServiceBuilder(item)
           .editSpec()
           .withClusterIP(old.getSpec().getClusterIP())
           .endSpec()
@@ -49,10 +45,5 @@ public class ServiceOperationsImpl extends BaseOperation<KubernetesClient, Servi
       } catch (Exception e) {
         throw KubernetesClientException.launderThrowable(e);
       }
-  }
-
-  @Override
-  public Service update(Service item) {
-      return replace(item);
   }
 }

